core/src/OpenMP/Kokkos_OpenMP_Instance.hpp

```diff
@@ -362,6 +365,48 @@ inline int OpenMP::impl_max_hardware_threads() noexcept {
   return Impl::g_openmp_hardware_max_threads;
 }
 
+namespace Experimental {
+namespace Impl {
+// Partitioning an Execution Space: expects space and integer arguments for
+// relative weight
+template <typename T>
+inline std::vector<OpenMP> create_OpenMP_instances(
+    OpenMP const& main_instance, std::vector<T> const& weights) {
+  static_assert(
+      std::is_arithmetic<T>::value,
+      "Kokkos Error: partitioning arguments must be integers or floats");
+  std::vector<OpenMP> instances(weights.size());
+  T total_weight = std::accumulate(weights.begin(), weights.end(), 0);
```

Comment:
```suggestion
  T total_weight = std::accumulate(weights.begin(), weights.end(), T(0));
```
otherwise result is an `int`.